src/browser_authentication_app_skeleton/src/actions/sign_ups/new.cr

```diff
@@ -1,5 +1,5 @@
 class SignUps::New < BrowserAction
-  include Auth::RedirectIfSignedIn
+  include Auth::RedirectSignedInUsers
```

Comment:
With all these renames, would it makes sense to put this in a higher level base controller? I know inheritance over compostability is heresy these days, but I feel like it makes sense to make some kind of `AuthenticatedAction` since most (but not all) of the app would use it.

Reply:
Or actually... can this just be the default in the browser action and instead you include `Auth::AllowGuests` to disable that behavior in the few cases that you need? Maybe that module would stub out the methods that prevent someone from logging in. Just thinking out loud here. I'm sure you've considered this much longer than I have 🙂

Reply:
@edwardloveall You can't see the change since it wasn't renamed, but in `BrowserAction` there is `Auth::RequireSignIn` by default. You only include these other ones when you want to change the default behavior which requires sign in. Is that kind of what you mean?

Reply:
I think I'm both under- and over-thinking it. I'll look at this again tomorrow and give you some better feedback 😴

Reply:
Ok, with fresh eyes I think this was just a silly mistake on my part. I was reading `RedirectSignedInUsers` as something that required users to sign in. It didn't make sense to me to include that functionality everywhere individually instead of having it exist in a base action.

And I think it does make sense to separate that from the AllowGuests mixin. Sorry! Disregard.